src/compiler/program.ts

```diff
@@ -2207,6 +2207,10 @@ namespace ts {
             addEmitFlags(importDecl, EmitFlags.NeverApplyImportHelper);
             setParent(externalHelpersModuleReference, importDecl);
             setParent(importDecl, file);
+            // explicitly unset the synthesized flag on these declarations so the checker API will answer questions about them
+            // (which is required to build the dependency graph for incremental emit)
+            (externalHelpersModuleReference as Mutable<Node>).flags ^= NodeFlags.Synthesized;
+            (importDecl as Mutable<Node>).flags ^= NodeFlags.Synthesized;

```

Comment:
I could be missing something, but does this actually unset the flag, or does it toggle? Seems like you might want

```suggestion
            (externalHelpersModuleReference as Mutable<Node>).flags &= ~NodeFlags.Synthesized;
            (importDecl as Mutable<Node>).flags &= ~NodeFlags.Synthesized;
```

Reply:
it "toggles" it, but it's guaranteed to be set, since these nodes come form the synthetic node factory.